core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala

```diff
@@ -386,6 +386,7 @@ class NodeToControllerRequestThread(
     if (response.authenticationException != null) {
       error(s"Request ${queueItem.request} failed due to authentication error with controller",
         response.authenticationException)
```

Comment:
Should we log: 
`Disconnecting the connection to the stale controller ${activeControllerAddress().map(_.idString).getOrElse("null")}`
like what we did for `NOT_CONTROLLER` error?

Reply:
Thanks for the review! Addressed in [c405f73](https://github.com/apache/kafka/pull/14760/commits/c405f7387bcfc94845689d46588c9fc75cbb8596)